core/chains/evm/forwarders/orm.go

```diff
@@ -1,105 +1,110 @@
 	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/utils/big"
-	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
 )
 
 //go:generate mockery --quiet --name ORM --output ./mocks/ --case=underscore
 
 type ORM interface {
-	CreateForwarder(addr common.Address, evmChainId big.Big) (fwd Forwarder, err error)
-	FindForwarders(offset, limit int) ([]Forwarder, int, error)
-	FindForwardersByChain(evmChainId big.Big) ([]Forwarder, error)
-	DeleteForwarder(id int64, cleanup func(tx pg.Queryer, evmChainId int64, addr common.Address) error) error
-	FindForwardersInListByChain(evmChainId big.Big, addrs []common.Address) ([]Forwarder, error)
+	CreateForwarder(ctx context.Context, addr common.Address, evmChainId big.Big) (fwd Forwarder, err error)
+	FindForwarders(ctx context.Context, offset, limit int) ([]Forwarder, int, error)
+	FindForwardersByChain(ctx context.Context, evmChainId big.Big) ([]Forwarder, error)
+	DeleteForwarder(ctx context.Context, id int64, cleanup func(tx sqlutil.Queryer, evmChainId int64, addr common.Address) error) error
+	FindForwardersInListByChain(ctx context.Context, evmChainId big.Big, addrs []common.Address) ([]Forwarder, error)
 }
 
-type orm struct {
-	q pg.Q
+type DbORM struct {
+	db sqlutil.Queryer
 }
 
-var _ ORM = (*orm)(nil)
+var _ ORM = &DbORM{}
+
+func NewORM(db sqlutil.Queryer) *DbORM {
+	return &DbORM{db: db}
+}
 
-func NewORM(db *sqlx.DB, lggr logger.Logger, cfg pg.QConfig) *orm {
-	return &orm{pg.NewQ(db, lggr, cfg)}
+func (o *DbORM) Transaction(ctx context.Context, fn func(*DbORM) error) (err error) {
+	return sqlutil.Transact(ctx, o.new, o.db, nil, fn)
 }
 
+// new returns a NewORM like o, but backed by q.
+func (o *DbORM) new(q sqlutil.Queryer) *DbORM { return NewORM(q) }
+
 // CreateForwarder creates the Forwarder address associated with the current EVM chain id.
-func (o *orm) CreateForwarder(addr common.Address, evmChainId big.Big) (fwd Forwarder, err error) {
+func (o *DbORM) CreateForwarder(ctx context.Context, addr common.Address, evmChainId big.Big) (fwd Forwarder, err error) {
 	sql := `INSERT INTO evm.forwarders (address, evm_chain_id, created_at, updated_at) VALUES ($1, $2, now(), now()) RETURNING *`
-	err = o.q.Get(&fwd, sql, addr, evmChainId)
+	err = o.db.GetContext(ctx, &fwd, sql, addr, evmChainId)
 	return fwd, err
 }
 
 // DeleteForwarder removes a forwarder address.
 // If cleanup is non-nil, it can be used to perform any chain- or contract-specific cleanup that need to happen atomically
 // on forwarder deletion.  If cleanup returns an error, forwarder deletion will be aborted.
-func (o *orm) DeleteForwarder(id int64, cleanup func(tx pg.Queryer, evmChainID int64, addr common.Address) error) (err error) {
+func (o *DbORM) DeleteForwarder(ctx context.Context, id int64, cleanup func(tx sqlutil.Queryer, evmChainID int64, addr common.Address) error) (err error) {
 	var dest struct {
 		EvmChainId int64
 		Address    common.Address
 	}
 
 	var rowsAffected int64
```

Comment:
Since the `rowsAffected` check got moved inside of the Transaction I don't think we need to define this var out here anymore.